### ISSUE TITLE
filter_next false positive

### ISSUE THREAD

**Issue by crumblingstatue** [rust-lang/rust-clippy#7561]
Lint name:
filter_next

I tried this code:

```rust
struct Foo;
impl Foo {
    fn filter(&self, _spec: i32) -> std::vec::IntoIter<i32> {
        unimplemented!()
    }
}

fn main() {
    let empty = Foo.filter(42).next().is_none();
    println!("{}", empty);
}
```

I expected to see this happen: Clippy doesn't suggest anything

Instead, this happened:
Clippy says this:
```
warning: called `filter(..).next()` on an `Iterator`. This is more succinctly expressed by calling `.find(..)` instead
 --> src/main.rs:9:17
  |
9 |     let empty = Foo.filter(42).next().is_none();
  |                 ^^^^^^^^^^^^^^^^^^^^^ help: try this: `Foo.find(42)`
  |
```

There is no find method on `Foo`.

### Meta

clippy 0.1.56 (ccffcaf 2021-08-11)

  ```
rustc 1.56.0-nightly (ccffcafd5 2021-08-11)
binary: rustc
commit-hash: ccffcafd55e58f769d4b0efc0064bf65e76998e4
commit-date: 2021-08-11
host: x86_64-unknown-linux-gnu
release: 1.56.0-nightly
LLVM version: 12.0.1
  ```

**Comment by xFrednet** [comment]
Thank you for the bug report :upside_down_face:. Seems like there is a check missing to make sure that the object actually implements `Iterator`.

@rustbot label +good-first-issue +I-suggestion-causes-error

**Comment by crumblingstatue** [comment]
That was a quick fix, thank you!
